iface/get.go

```diff
@@ -2,79 +2,176 @@ package iface
 
 import (
 	"fmt"
-	"go/token"
+	"go/ast"
 	"go/types"
+	"path/filepath"
 	"sort"
 	"strings"
 
-	"github.com/pkg/errors"
 	"golang.org/x/tools/go/packages"
 )
 
-func GetInterface(dir, ifaceName string) (Interface, error) {
-	cfg := &packages.Config{Mode: packages.LoadSyntax}
-	pkgs, err := packages.Load(cfg, dir)
-	if err != nil {
-		return Interface{}, errors.Wrap(err, "error loading package info")
-	}
+// fileInfo represents syntactic and type information for a file containing
+// interfaces to be mocked.
+type fileInfo struct {
+	pkg      *packages.Package
+	fileNode *ast.File
+	objects  []types.Object
+}
+
+// GetAllInterfaces searches the given packages for interfaces annotated with a
+// "go:mock" directive, returning text-template-friendly representations grouped
+// by output file.
+func GetAllInterfaces(pkgs []*packages.Package) (map[string]File, error) {
+	fileInfoByPath := map[string]*fileInfo{}
+	for _, pkg := range pkgs {
+		for _, fileNode := range pkg.Syntax {
+			ast.Inspect(fileNode, func(node ast.Node) bool {
+				// Only consider declarations with godoc comments.
+				decl, isGenDecl := node.(*ast.GenDecl)
+				if !isGenDecl || decl.Doc == nil {
```

Comment:
Is it worth putting an explicit check for a `node = nil` here? I think you get saved by is `isGenDecl` being false in that scenario, but my understanding is nil is the base-case parameter

Reply:
Certainly doesn't hurt: b37e654d297804fd0d245fdfbdb8e46afa4ac4b9